### ISSUE TITLE
Why is this user.js scripted matched?

### ISSUE THREAD

**Issue by tobwz** [erosman/support#308]
I setup (among others) a *.user.js script in Firemonkey with the content shown at the bottom of this post.
It should match ONLY webpages from Facebook.

Now when I load certain webpages like e.g. the following:

https://www.ilovefreesoftware.com/06/featured/free-decision-table-maker-software-for-windows.html

then this Facebook script is matched (too).
Maybe I am blind: But why is the Facebook script matched here?
When I click on the Firemonkey toolbar icon it appears there.

It doesn't change the content on the mentioned ilovefreesoftware.com web page but maybe it could have unwanted side effected for other non-Facebook domains.

This bug (?) occurs in approx 3% of all webpages.
I am using Firefox v87 64bit on Win7

```
// ==UserScript== 
// @name            ChangeLayoutFacebook
// @namespace       nsChangeLayoutFacebook
// @description     Change Layout facebook.com
// @match           https://*.facebook.com/*
// @version         1.2.0
// @grant           GM_addStyle
// @require         http://code.jquery.com/jquery-latest.js
// ==/UserScript== 

$("._3ob9").remove();
GM_addStyle("_62uh { display: none !important; }");
GM_addStyle("#pagelet_growth_expanding_cta { display: none !important; }");
```

**Comment by erosman** [comment]
The mentioned page has a facebook iframe.
However, in my version of FireMonkey, it doesnt match the above page. Which version are you using?

BTW, the above userScript would work a lot more efficiently as a userCSS and will apply to dynamically created elements that are created later.

```css
/*
==UserCSS== 
@name            ChangeLayoutFacebook
@namespace       nsChangeLayoutFacebook
@description     Change Layout facebook.com
@match           https://*.facebook.com/*
@version         1.2.0
==/UserCSS== 
*/

._3ob9, _62uh, #pagelet_growth_expanding_cta {
  display: none !important;
}
```

AFA userScript & `GM_addStyle()`, it is more efficient to run it once e,g,

```js
const css = `
_62uh { 
  display: none !important; 
}
#pagelet_growth_expanding_cta { 
  display: none !important; 
}`;

GM_addStyle(css);
```

**Comment by tobwz** [comment]
I am using Firemonkey v2.22

**Comment by tobwz** [comment]
iframe matching could be the reason....interesting

How can I avoid iframe matching?

**Comment by erosman** [comment]
I know what is going on. The iframe is activated with JavaScript. I have JavaScript blocked so it didnt show active for me. After enabling JS on above page, it shows the userScript as active.

**Note:** Although the display shows it is active, the script actually is not injected in your example page.

I will fix it and release a new upgrade.

**Comment by erosman** [comment]
Fixed.. it will be in v2.23

Thank you for posting the bug :+1:

**Comment by pintassilgo** [comment]
> How can I avoid iframe matching?

Add **\@noframes** in metadata block.

(As I already requested, I think FM should support user custom **\@noframes** per script, to prevent editing non personal scripts)

**Comment by erosman** [comment]
The problem is fixed in v2.23 so you don't have to do anything. Please wait for v2.23.

FireMonkey _default_ is `@noframes` or `@allFrames: false`.

**Comment by erosman** [comment]
v2.23 uploaded ... let me know if there are any issues
